rfcs/0026-anyone-can-pay/0026-anyone-can-pay.md

```diff
@@ -0,0 +1,270 @@
+---
+Number: "0026"
+Category: Standards Track
+Status: Proposal
+Author: Xuejie Xiao
+Organization: Nervos Foundation
+Created: 2020-09-03
+---
+
+# Anyone-Can-Pay Lock
+
+This RFC describes a new lock script for CKB that can accept any amount of [Simple UDT](../0025-simple-udt/0025-simple-udt.md) or CKB payment. Previously, one can only transfer to another user at least 61 CKBytes when using the default lock, possibly more when using other lock scripts or type scripts. This is becoming a bigger problem when UDT support lands in CKB: a naive UDT transfer operation will not only require UDTs, but CKByte to keep the UDTs in a cell as well.
+
+Here we try to solve the problem by introducing a new anyone-can-pay lock script, which can be unlocked not only by the validation of a signature, but also by accepting any amount of payment. This way, a user should be able to send any amount of CKBytes or UDTs to a cell using anyone-can-pay lock instead of always creating a new cell. It thus provides a solution to both problems above.
+
+## Script Structure
+
+The anyone-can-pay lock is built upon the default secp256k1-blake2b-sighash-all lock with additions to the script args part. The new anyone-can-pay lock can accept any of the following script args format:
+
+```
+<20 byte blake160 public key hash>
+<20 byte blake160 public key hash> + <1 byte CKByte minimum>
+<20 byte blake160 public key hash> + <1 byte CKByte minimum> + <1 byte UDT minimum>
```

Comment:
There’s no `+` in the following examples, maybe they also can be removed here.

The RFC does not talk about how the two minimums are encoded.

Reply:
The minimums may lead to integer overflow, The RFC has to specify the maximum allowed values. From the implementation one uses u64 and another uses u128

```
  uint64_t min_ckb_amount;
  uint128_t min_udt_amount;
```

Reply:
Added a new comment on this line https://github.com/nervosnetwork/rfcs/pull/205/files/722dba240f8c33338a67a844000c691fcb51a979#r483358842

Reply:
Personally I feel like it could be up to the contracts to handle integer overflows. There might be an implementation that choose to implement arbitrary long integer arithmetic, which won't have overflowing problems.

Reply:
This explanation can be added to the RFC explicitly, however, I don’t have the strong opinion to block the PR because of this.